src/test/kotlin/build/buf/gradle/BreakingWithProtobufGradleTest.kt

```diff
@@ -14,16 +14,22 @@
 
 package build.buf.gradle
 
+import org.junit.Ignore
 import org.junit.jupiter.api.Test
 import java.nio.file.Paths
 
 class BreakingWithProtobufGradleTest : AbstractBreakingTest() {
     override fun protoFile() = Paths.get(protoDir.path, "buf", "test", "v1", "test.proto")
 
     @Test
+    @Ignore
+    override fun `normally breaking schema with an ignore`() {
```

Comment:
Modifying tests for v2 and disabling them where they fail removes verification of a currently valid workflow. Instead I think it makes sense to build a new set of tests for v2 in a v2 package (and move the existing tests to a v1 package). Then we can copy over all of the existing tests and handle the failures on a case-by-case basis without losing verification of existing behavior. If needed, the buf tool version can be injected with a Gradle project property on a per-test basis if a new tool version fails for existing configs.

In particular I think instead of disabling those tests we should codify the failures with assertions on exactly how they fail so that inadvertent changes don't get ignored. Ideally, of course, we identify which cases are still valid and which are going to fall by the wayside, hopefully with a migration plan for existing users we can add to the readme.

One advantage of this approach is that I don't have to re-contextualize myself with the plugin all at once after at least a year away from it 😄.

Reply:
It might be a lot to maintain multiple sets of tests and to try to support all combinations of plugin version and Buf version. For now any tests that can continue to pass using `v1` config files are still present. For the case of using the Google protobuf plugin alongside a `buf.yaml` file, I am going to try to find a way to make it work by generating a suitable `buf.yaml` file rather than a `buf.work.yaml` file but I am planning to tackle that in a later PR.

Reply:
If you can point me to how we can set up a way to test just the specific v1-only cases with an injected Buf version, I'm open to doing that.

Reply:
It'll be a lot, but I think that's the most important thing the test suite can do, and it's how I'd personally prefer things get set up. It's the only way to prove nothing breaks!

Here's specification of a dynamic version in a test build.gradle file: https://github.com/bufbuild/buf-gradle-plugin/blob/main/src/test/kotlin/build/buf/gradle/AbstractBufIntegrationTest.kt#L69

The same can be done for the Buf tool version in the plugin's Extension.

Reply:
Retrieving that version from a Kotlin build file: https://github.com/bufbuild/buf-gradle-plugin/blob/main/src/test/resources/GenerateTest/generate_java_with_kotlin_dsl/build.gradle.kts#L21

From Groovy: https://github.com/bufbuild/buf-gradle-plugin/blob/main/src/test/resources/BreakingWithProtobufGradleTest/breaking_schema_fails_with_latest_release_and_previousversion/build.gradle#L3